build-chunk.sh

```diff
@@ -54,7 +41,7 @@ function extract_variants() {
 ARGS=$(getopt -o h,:c:n: --long help:,chunk:,name: -- "$@")
 eval set -- "${ARGS}"
 
-[ ! -f ${BACKUP_FILE} ] && echo "${YELLOW}Creating a backup of ${ORIGINAL_FILE} as it does not exist yet...${NC}" && cp ${ORIGINAL_FILE} ${BACKUP_FILE}
+save_original
```

Comment:
Might want to consider adding `trap "restore_original" ERR` immediately after `save_original`. This way, if some error happens, we exit with a predictable dazzle.yaml file.

To test, try running `./build-chunk.sh -c lang-go`, which results in an error because no variant is specified. If you do that, it leaves behind an altered `dazzle.yaml`.

Reply:
I am now trapping EXIT, which means we will restore the original regardless of what the reason for the exit was. This also means we do not need to call restore_original explicitly at the end because it will be called automatically. The disadvantage is that it will only work with bash, but this should be ok.

Reply:
Awesome, thank you!